### PR TITLE
feat(core/react): add free layout

### PR DESCRIPTION
**Pull request by jinphic**
add FreeLayout feature.

Add hooks and handlers to achieve this function. 
When the component has freeLayout properties in designerProps, it show a button with text 'Free' on the top left corner. Developer can press the button, drag and drop to place the component everywhere on the designer panel.

Before submitting a pull request, please make sure the following is done...

- [x] Ensure the pull request title and commit message follow the Commit Specific in English.
- [x] Fork the repo and create your branch from main.
- [ ] If you've added code that should be tested, add tests!
- [ ] If you've changed APIs, update the documentation.
- [ ] Ensure the test suite passes (npm test).
- [x] Make sure your code lints (npm run lint) - we've done our best to make sure these rules match our internal linting guidelines.

### CODE REVIEW

**Comment by jinphic** [comment]
> add FreeLayout feature.
> 
> Add hooks and handlers to achieve this function. When the component has freeLayout properties in designerProps, it show a button with text 'Free' on the top left corner. Developer can press the button, drag and drop to place the component everywhere on the designer panel.
> 
> Before submitting a pull request, please make sure the following is done...
> 
> * [x]  Ensure the pull request title and commit message follow the Commit Specific in English.
> * [x]  Fork the repo and create your branch from main.
> * [ ]  If you've added code that should be tested, add tests!
> * [ ]  If you've changed APIs, update the documentation.
> * [ ]  Ensure the test suite passes (npm test).
> * [x]  Make sure your code lints (npm run lint) - we've done our best to make sure these rules match our internal linting guidelines.

image

**Comment by janryWang** [comment]
感觉改名traslatable要优雅一点

**Comment by janryWang** [comment]
CLA签一下

**Comment by jinphic** [comment]
> 感觉改名 freeLayout --> traslatable horizontal --> x vertial --> y 是不是更优雅简洁一些

可以， 我改一下提个新的pr。我也觉得函数名字太长了

**Comment by jinphic** [comment]
> CLA签一下

嗯嗯，我提新的pr的时候注意一下

**Comment by jinphic** [comment]
> 感觉改名 freeLayout --> traslatable horizontal --> x vertial --> y 是不是更优雅简洁一些

这里修改好了

**Comment by jinphic** [comment]
> CLA签一下

You have agreed to the CLA for alibaba/designable
签CLA协议的网页提示我已经签了。但是我在这个pr页面还是提示`CLA not signed yet`。这里该怎么操作

**Comment by qiaoyuwen** [comment]
> > CLA签一下
> 
> You have agreed to the CLA for alibaba/designable 签CLA协议的网页提示我已经签了。但是我在这个pr页面还是提示`CLA not signed yet`。这里该怎么操作

你commit的邮箱和你github账号没关联

File: packages/react/src/widgets/AuxToolWidget/TranslateHandler.tsx [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+import { useDesigner, usePrefix } from '../../hooks'
+import { TreeNode } from '@designable/core'
+
+export interface ITranslateHandlerProps {
+  node: TreeNode
+}
+
+export const TranslateHandler: React.FC<ITranslateHandlerProps> = (props) => {
+  const designer = useDesigner()
+  const prefix = usePrefix('aux-node-free-layout-handler')
```
Comment (janryWang) [inline packages/react/src/widgets/AuxToolWidget/TranslateHandler.tsx:12]: 改名translate-handler

**Comment by janryWang** [comment]
建议重新提个PR，commit的邮箱是腾讯内网邮箱

**Comment by jinphic** [comment]
> 建议重新提个PR，commit的邮箱是腾讯内网邮箱

好的，麻烦你close了，我晚上回去换台电脑再来一个

**Pull request closed without merging**